app/frontend/src/features/userQueries/useUsers.ts

```diff
@@ -33,7 +33,7 @@ export default function useUsers(ids: number[], invalidate: boolean = false) {
 
   const usersById = isLoading
     ? undefined
-    : new Map(queries.map((q) => [q.data!.userId, q.data!]));
+    : new Map(queries.map((q, index) => [ids[index], q.data]));
```

Comment:
This now supports partial data, so some ids can be undefined. It assumes that the order of queries stays consistent though, I don't see why that wouldn't be the case.

Reply:
I suppose it's possible the user has new friend requests accepted in between, in which case this hook could be called again with a different list of ids?

Reply:
That should be fine, I mean more that calling `useQueries(ids.map(...` uses the ids in order, and returns a list of queries where the query keys are the same order. Because there is no way to get the query key from the query result object.

Reply:
Yeah I think it should be fine too, as the array of `QueryObserverResult` seems to come back in the same order as the list passed into `useQueries`.